src/node_file.cc

```diff
@@ -1641,6 +1659,24 @@ static void Fsync(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+static void FsyncSync(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  const int argc = args.Length();
+  CHECK_GE(argc, 1);
+
+  const int fd = GetValidatedFd(env, args[0]);
+  if (fd == (1 << 30)) return;
```

Comment:
What is the purpose of this line?

Reply:
This line is why we have 6000% improvement. We skip `uv_fs_fsync` for the benchmark input. We should remove it

Reply:
When an Exception is thrown in `GetValidatedFd()`, an Invalid value of `1<<30` is returned. Since it throws an Exception, wouldn't there be any need for further action?

Reply:
Yeah. Makes sense!